components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -196,6 +196,12 @@ void AIChatTabHelper::OnTabContentRetrieved(int64_t for_navigation_id,
     VLOG(1) << __func__ << ": No data";
     return;
   }
+
+  // tokens + max_new_tokens must be <= 4096 (llama2)
+  // 8092 chars, ~3,098 tokens (reserved for article)
+  // 1k chars, ~380 tokens (reserved for prompt)
+  contents_text = contents_text.substr(0, 8092);
+
```

Comment:
(and here we can set a bool flag to indicate to the user that only a part of the page is being shared with leo as it's too long, in the subsequent PR)